controllers/awsadapterconfig_controller.go

```diff
@@ -84,20 +92,44 @@ func (r *AWSAdapterConfigReconciler) Reconcile(ctx context.Context, req ctrl.Req
 		return r.updateLastPollStatusFailure(ctx, objOld, "error occurred while loading aws sdk config", err, &l, time.Now())
 	}
 	l.Info("AWS SDK config loaded successfully")
-	svc := eks.NewFromConfig(cfg)
+
+	stsClient := sts.NewFromConfig(cfg)
+	ec2Client := ec2.NewFromConfig(cfg)
+	ecrClient := ecr.NewFromConfig(cfg)
+	eksClient := eks.NewFromConfig(cfg)
+	inspector2Client := inspector2.NewFromConfig(cfg)
 
 	objNew := objOld.DeepCopy()
 	objNew.Status.EKSCluster = &securityv1alpha1.EKSCluster{}
+	objNew.Status.AccountData = &securityv1alpha1.AccountData{}
 
 	clusterFound := false
 
-	if x, err := svc.ListClusters(context.TODO(), &eks.ListClustersInput{}); err == nil {
+	if callerIdentity, err := stsClient.GetCallerIdentity(ctx, &sts.GetCallerIdentityInput{}); err == nil && callerIdentity.Account != nil {
+		objNew.Status.AccountData.ID = callerIdentity.Account
+
+		x, err := inspector2Client.BatchGetAccountStatus(ctx, &inspector2.BatchGetAccountStatusInput{AccountIds: []string{*callerIdentity.Account}})
+		if err == nil {
+			objNew.Status.AccountData.InspectorEnabledEC2 = aws.Bool(x.Accounts[0].ResourceState.Ec2.Status == inspector2Types.StatusEnabled)
+			objNew.Status.AccountData.InspectorEnabledECR = aws.Bool(x.Accounts[0].ResourceState.Ecr.Status == inspector2Types.StatusEnabled)
```

Comment:
There will always be at least one account if BatchGetAccountStatus does not return error? Or array overflow?